src/Function.cpp

```diff
@@ -77,6 +77,53 @@ static long cur_func_idx;				// Index into FuncList that we are currently workin
 static bool param_first=true;			// Flag to track output of commas
 static int builtin_functions_cnt;
 
+void
+FunctionAttribute::OutputAttribute(std::ostream &out)
+{
+	out << " __attribute__((" << attribute << "))";
+}
+
+BooleanFunctionAttribute::BooleanFunctionAttribute(int prob, string name)
```

Comment:
Please use base class constructor here: ` : FunctionAttribute (name, prob)`

Reply:
Got it. Using base constructor wherever required.